docs/modules/ROOT/pages/reference/extensions/google-pubsub.adoc

```diff
@@ -32,3 +32,12 @@ Please refer to the above link for usage and configuration details.
 ----
 
 Check the xref:user-guide/index.adoc[User guide] for more information about writing Camel Quarkus applications.
+
+== Camel Quarkus limitations
+
+By default, the Camel PubSub component uses JDK object serialization via `ObjectOutputStream` whenever the message body is anything other than `String` or `byte[]`.
+
+Since such serialization is not yet supported by GraalVM, this extension provides a custom Jackson based serializer to serialize complex message payloads as JSON.
```

Comment:
The custom Jackson based serializer is used in both JVM and native modes (the text seems to suggest that).

BTW, we'll see by tomorrow, whether serialization will be supported in GraalVM 21 https://github.com/oracle/graal/pull/2730#issuecomment-737135245

Reply:
> The custom Jackson based serializer is used in both JVM and native modes

Yes, it's better IMO to be consistent and have the same behaviour in both modes.